drivers/unix/file_access_unix.cpp

```diff
@@ -137,11 +138,32 @@ Error FileAccessUnix::open_internal(const String &p_path, int p_mode_flags) {
 	return OK;
 }
 
+void FileAccessUnix::_sync() {
+	ERR_FAIL_NULL(f);
+
+	fflush(f);
+	int fd = fileno(f);
+	ERR_FAIL_COND(fd < 0);
+
+#ifdef __APPLE__
+	fcntl(fd, F_BARRIERFSYNC);
+#else
+	int fsync_error;
+	do {
+		fsync_error = fsync(fd);
+	} while (fsync_error < 0 && errno == EINTR);
+	ERR_FAIL_COND_MSG(fsync_error < 0, strerror(errno));
```

Comment:
In case of failure it probably should set `last_error` as well.

Reply:
Unfortunately, `ERR_FAIL_COND_MSG` doesn't seem to be easily compatible with `last_error`. I could check the condition twice, once before `ERR_FAIL_COND_MSG` to set `last_error` and once within, though that's a bit awkward.

However, I noticed that elsewhere in this file, the paradigm for `last_error` is that it represents usage errors, such as hitting the end of the file, being unable to find the file, or not having permission to open the file. Internal errors, such as failing to get the file length or a rename failure, emit an error message but do not set `last_error`. My default position would be to remain consistent with the rest of the file unless your intention is to start deliberately changing existing semantics.

Given that, would you like me to go ahead and make the change?